packages/jest-dom-mocks/src/dimension.ts

```diff
@@ -0,0 +1,68 @@
+enum SupportedDimension {
+  OffsetWidth = 'offsetWidth',
+  OffsetHeight = 'offsetHeight',
+  ScrollWidth = 'scrollWidth',
+  ScrollHeight = 'scrollHeight',
+}
+
+type DimensionMap = {[T in SupportedDimension]: HTMLElement | Element};
+type MockedDimensions = {[T in SupportedDimension]: number};
+
+export default class Dimension {
+  private isUsingMock = false;
+  private mocks: Partial<MockedDimensions> = {};
+  private dimensionMap: DimensionMap = {
+    [SupportedDimension.OffsetWidth]: HTMLElement.prototype,
+    [SupportedDimension.OffsetHeight]: HTMLElement.prototype,
+    [SupportedDimension.ScrollWidth]: Element.prototype,
+    [SupportedDimension.ScrollHeight]: Element.prototype,
+  };
+
+  mock(mocks: Partial<MockedDimensions>) {
```

Comment:
Do you think it's worth making this mock a little more flexible at this point? Something like also accepting a function that takes an element and the property being measured and returns a number (for cases where multiple elements are measured)?

Reply:
Yeah, not a bad idea, will change 👍